isisd/isis_spf_delay.c

```diff
@@ -0,0 +1,480 @@
+/*
+ * This is an implementation of the IETF SPF delay algorithm 
+ * as explain in draft-ietf-rtgwg-backoff-algo
+ *
+ * Module name: isis_spf_delay.c
+ * Version:     0.1
+ * Created:     25-01-2017 by S. Litkowski 
+ * Copyright (C) 2017 Orange Labs http://www.orange.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
+ * MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+#include <sys/time.h>
+
+#include "linklist.h"
+#include "thread.h"
+#include "vty.h"
+#include "stream.h"
+#include "memory.h"
+#include "log.h"
+#include "prefix.h"
+#include "command.h"
+#include "hash.h"
+#include "if.h"
+#include "checksum.h"
+#include "md5.h"
+#include "table.h"
+
+#include "isisd/dict.h"
+#include "isisd/isis_constants.h"
+#include "isisd/isis_common.h"
+#include "isisd/isis_flags.h"
+#include "isisd/isis_circuit.h"
+#include "isisd/isisd.h"
+#include "isisd/isis_tlv.h"
+#include "isisd/isis_lsp.h"
+#include "isisd/isis_pdu.h"
+#include "isisd/isis_dynhn.h"
+#include "isisd/isis_misc.h"
+#include "isisd/isis_csm.h"
+#include "isisd/isis_adjacency.h"
+#include "isisd/isis_spf.h"
+
+#include "lib/thread.h"
+
+#include "isis_spf_delay.h"
+#include <time.h>
+#include <malloc.h>
+#include <stdio.h>
+
+const char       *isis_spf_delay_states[3] = { "QUIET", 
+					       "SHORT_WAIT", 
+					       "LONG_WAIT"};
+
+char *
+str_mtimestr(time_t uptime)
+{
+
+#define ONE_MINUTE_SECOND 60*60
+#define ONE_DAY_SECOND 60*60*24
+#define ONE_WEEK_SECOND 60*60*24*7
+
+  struct tm *tm;
+  char *buf = (char *)malloc(15);
```

Comment:
As mentioned in another comment, I finally removed this function as I moved next_spf to monotonic time.